### ISSUE TITLE
Filter does not work with realtime streaming

### ISSUE THREAD

**Issue by bjedrzejewski** [spring-cloud-stream-app-starters/twitter#8]
I have tried to use your starter app and I found the following issue:

I set up all my secrets correctly and then add to the stream:

`--track=java --stream-type=filter`

What I end up getting is 5-6 filtered tweets (correctly with Java keyword), but after that the data stops flowing. The `sample` stream type works fine. Could you help?

**Comment by bjedrzejewski** [comment]
I will add- I get no errors in any of the application logs (I checked in the respective directories).

**Comment by tzolov** [comment]
@bjedrzejewski in order to establish a base line, could you please run native filter.json API (e.g. without SCDF) and share what response do you get? 
Perhaps you can try with the https://apigee.com/console/twitter ?

**Comment by tzolov** [comment]
Ok, i believe i can confirm the issue. When using the [spring-social-twitter](http://projects.spring.io/spring-social-twitter/) directly:
```java
new TwitterTemplate(consKey, consSecret, accessToken, accessTokenSecret)
  .streamingOperations()
  .filter("java", Arrays.asList(new StreamListener() {
		public void onTweet(Tweet tweet) {System.out.println(tweet.getText());}
		public void onDelete(StreamDeleteEvent streamDeleteEvent) {}
		public void onLimit(int i) {}
		public void onWarning(StreamWarningEvent streamWarningEvent) {}
}));
```
I can generate a continuous stream of tweets. But with `twittersouce` produces only few tweets for the same `track` parameter. 

I will investigate further

**Comment by bjedrzejewski** [comment]
@tzolov thanks so much for looking into that! It is a great feature for teaching people Data Flow, so would be nice to have it working!

**Comment by sabbyanandan** [comment]
Hi, @bjedrzejewski: Christian's spring-cloud-stream-app-starters/twitter#9 addresses the eventual hangup. There was an unnecessary `break` statement that caused it. Once reviewed and merged, we will release 1.3.2.RELEASE of this application. 

Please try the PR and let us know if you see an anomaly.

**Comment by bjedrzejewski** [comment]
Thanks guys, I will try the PR tonight!
